### PR TITLE
Macro string functions

### PR DESCRIPTION
**Pull request by cedporter**
Don't actually need to mark the contents of these macro string functions as strings. Do need to ensure that unmatched quotes within them are not erroneously identified as the start and end of strings. Also improved macro variable identification.

### CODE REVIEW

File: components/prism-sas.js [hunk 1]
```diff
@@ -245,19 +255,14 @@
 					pattern: /%(?:NRBQUOTE|NRQUOTE|NRSTR|BQUOTE|QUOTE|STR)/i,
 					alias: 'keyword'
 				},
-				'string': {
-					pattern: /(\()[^)]+/,
-					lookbehind: true
+				'macro-keyword': macroKeyword,
+				'macro-variable': macroVariable,
+				'escaped-char': {
+					pattern: /%(?:'|"|\(|\)|<|>|=|¬|\^|~|;|,|#)/i,
```
Comment (RunDevelopment) [inline components/prism-sas.js:261]:
```suggestion
					pattern: /%['"()<>=¬^~;,#]/,
```

File: components/prism-sas.js [hunk 2]
```diff
@@ -8,6 +8,16 @@
 		alias: 'number'
 	};
 
+	var macroVariable = {
+		pattern: /&[a-z_][a-z_1-9]*/i
```
Comment (RunDevelopment) [inline components/prism-sas.js:12]: `0` is not allowed?

Comment (cedporter) [reply components/prism-sas.js]: It is allowed. Thanks for catching that.

**Comment by RunDevelopment** [comment]
Thank you for contributing!

**Pull request merged**